src/integrations/prefect-dbt/tests/test_utilities.py

```diff
@@ -0,0 +1,77 @@
+import os
+from pathlib import Path
+
+import pytest
+import yaml
+from prefect_dbt.utilities import get_profiles_dir, load_profiles_yml
+
+SAMPLE_PROFILES = {
+    "jaffle_shop": {
+        "outputs": {
+            "dev": {
+                "type": "duckdb",
+                "path": "jaffle_shop.duckdb",
+                "schema": "main",
+                "threads": 4,
+            }
+        }
+    }
+}
+
+
+@pytest.fixture
+def temp_profiles_dir(tmp_path):
+    profiles_dir = tmp_path / ".dbt"
+    profiles_dir.mkdir()
+
+    profiles_path = profiles_dir / "profiles.yml"
+    with open(profiles_path, "w") as f:
+        yaml.dump(SAMPLE_PROFILES, f)
+
+    return str(profiles_dir)
+
+
+def test_get_profiles_dir_default():
+    if "DBT_PROFILES_DIR" in os.environ:
+        del os.environ["DBT_PROFILES_DIR"]
+
+    expected = os.path.expanduser("~/.dbt")
+    assert get_profiles_dir() == expected
+
+
+def test_get_profiles_dir_from_env():
+    test_path = "/custom/path"
+    os.environ["DBT_PROFILES_DIR"] = test_path
+    try:
+        assert get_profiles_dir() == test_path
+    finally:
+        del os.environ["DBT_PROFILES_DIR"]
```

Comment:
You can use `monkeypatch.setenv` to simplify this a smidge:
```suggestion
def test_get_profiles_dir_from_env(monkeypatch):
    test_path = "/custom/path"
    monkeypatch.setenv("DBT_PROFILES_DIR", test_path)
    assert get_profiles_dir() == test_path
```